ui/src/app/models/workflowRequest.model.ts

```diff
@@ -16,37 +16,103 @@
 import { WorkflowEntryModel } from './workflowEntry.model';
 import { JobEntryModelObject } from './jobEntry.model';
 import set from 'lodash/set';
+import { WorkflowJoinedModel } from './workflowJoined.model';
+import { PropertiesModel, SensorModel } from './sensor.model';
+import { DagDefinitionJoinedModel } from './dagDefinitionJoined.model';
+import { JobDefinitionModel, JobParametersModel } from './jobDefinition.model';
+
+class WorkflowDetails {
+  constructor(
+    public name?: string,
+    public isActive?: boolean,
+    public project?: string,
+    public created?: Date,
+    public id?: number,
+    public updated?: Date,
+  ) {}
+}
+
+class WorkflowSensor {
+  constructor(public workflowId?: number, public sensorType?: { name: string }, public properties?: PropertiesModel, public id?: number) {}
+}
+
+class WorkflowJobDefinition {
```

Comment:
Why did you duplicate this from `jobDefinition.model.ts`?

Reply:
WorkflowDetails, WorkflowSensor, and WorkflowJobDefinition are duplicates. To make mapping and use generic data, I need class copies with all optional fields.( I can create empty object and set it dynamically and do the mapping.)

Reply:
Ah I see. This can probably be remedied by using plain objects. I'll try to remove the duplicates in #168